### PR TITLE
MAST: Add kepler to cloud accessable missions

### PR DESCRIPTION
**Pull request by ceb8**
The main change in the PR is the addition of the Kepler mission to missions accessible in the cloud. I also made a number of smaller improvements to the cloud access functionality and a few code style updates.

Specifically:
- Added the `kepler_paths` function
- Added Kepler as an option to the `paths` and `has_paths` functions
- Updated the  `get_cloud_uri` function so that if a file is not found it produces a warning and returns None rather than throwing an exception
- Added a cloud_only flag to the `download_products` function to prevent fallback to MAST when a file is not found in the cloud
- Updated the Cloud Access documentation to be more clear and include examples of using the `download_products` function
- Improved existing and added new docstrings for cloud functions as needed
- Generally made small changes to correct typos etc in existing docstrings throughout module
- Pep8-ified variable names in fpl.py
- Changed all instances of % formatting to .format()

### CODE REVIEW

File: astroquery/mast/core.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _prepare_service_request_string(json_obj):
 def _parse_type(dbtype):
     """
     Takes a data type as returned by a database call and regularizes it into a
-    triplet of the form (human readable datatype, python datatype, default value).
+    triplet of the form (human readable datatype, python datatype, defult value).
```
Comment (bsipocz) [inline astroquery/mast/core.py]: remove change :)

File: astroquery/mast/core.py [hunk 2]
```diff
@@ -1730,18 +1741,22 @@ def progress_callback(numbytes):
             bkt.download_file(bucket_path, local_path, ExtraArgs={"RequestPayer": "requester"},
                               Callback=progress_callback)
 
-    def _download_files(self, products, base_dir, cache=True):
+    def _download_files(self, products, base_dir, cache=True, cloud_only=False,):
         """
-        Takes an `astropy.table.Table` of data products and downloads them into the dirctor given by base_dir.
+        Takes an `~astropy.table.Table` of data products and downloads them into the dirctory given by base_dir.
```
Comment (bsipocz) [inline astroquery/mast/core.py]: typo

File: astroquery/mast/core.py [hunk 3]
```diff
@@ -1730,18 +1741,22 @@ def progress_callback(numbytes):
...
-        products : `astropy.table.Table`
+        products : `~astropy.table.Table`
             Table containing products to be downloaded.
         base_dir : str
             Directory in which files will be downloaded.
         cache : bool
-            Default is True. If file is found on disc it will not be downloaded again.
+            Default is True. If file is found on disk it will not be downloaded again.
+        cloud_only : bool, optional
+            Default False. If set to True and cloud data access is enabled (see `enable_cloud_dataset`)
```
Comment (bsipocz) [inline astroquery/mast/core.py]: add double quotes, don't think otherwise sphinx will be happy

File: astroquery/mast/core.py [hunk 4]
```diff
@@ -1812,10 +1833,13 @@ def download_products(self, products, download_dir=None,
             will be downloaded that can be used to download the data files at a later time.
         mrp_only : bool, optional
             Default False. When set to true only "Minimum Recommended Products" will be returned.
+        cloud_only : bool, optional
+            Default False. If set to True and cloud data access is enabled (see `enable_cloud_dataset`)
```
Comment (bsipocz) [inline astroquery/mast/core.py]: double quotes

File: astroquery/mast/core.py [hunk 5]
```diff
@@ -1812,10 +1833,13 @@ def download_products(self, products, download_dir=None,
...
+            files that are not found in the cloud will be skipped rather than downloaded from MAST
+            as is the default behavior. If cloud access is not enables this argument as no affect.
         **filters :
-            Filters to be applied.  Valid filters are all products fields listed
-            `here <https://masttest.stsci.edu/api/v0/_productsfields.html>`__ and 'extension'
-            which is the desired file extension.
+            Filters to be applied.  Valid filters are all products fields returned by 
+            `get_metadata` with query_type = 'products' and 'extension' which is the desired file extension.
```
Comment (bsipocz) [inline astroquery/mast/core.py]: double quote. Maybe do the full ``get_metadata`` call as an example?

File: astroquery/mast/core.py (hunk 2 above)
Comment (ceb8) [reply astroquery/mast/core.py]: I half fixed it..... ;)

File: astroquery/mast/core.py (hunk 3 above)
Comment (ceb8) [reply astroquery/mast/core.py]: Where are you suggesting I add double quotes?

**Comment by ceb8** [comment]
@bsipocz I'm looking at the travis errors, and it seems like they are coming from sphinx warnings, specifically of the references to `requests.Response` in doc strings.  On investigating our current docs that link does indeed fail, it attempts to resolve to "https://2.python-requests.org//en/stable/api/#requests.Response" which gives me a secure connection failed in all browsers I try it in.  If I change the "https" to "http" it easily redirects to "https://requests.kennethreitz.org//en/stable/api/#requests.Response" which makes me think that requests has let some cert expire or something. Although I don't really understand how the resolution from python module name to doc url works so... In any case, I'm not sure how to fix this error, it seems like it's an upstream problem.

**Comment by bsipocz** [comment]
@ceb8 - indeed, that's upstream and has been fixed in master.

**Comment by ceb8** [comment]
@bsipocz Great! I will do a rebase then!

File: astroquery/mast/core.py (hunk 3 above)
Comment (bsipocz) [reply astroquery/mast/core.py]: never mind me, these are available in the API docs as links (as they are defined in the same namespace). Not sure where my mind was when I did the review 😅

**Comment by bsipocz** [comment]
No need to rebase, this built nicely locally.

**Comment by bsipocz** [comment]
I'm not sure why the bot complains about the changelog, everything looks OK to me.

**Comment by bsipocz** [comment]
Thanks @ceb8!

**Pull request merged**